### PR TITLE
Feat/replace sanitizing

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

To improve security (against XSS Attacks) custom "sanitiser" was replaced with [DOMPurify](https://github.com/cure53/DOMPurify).

# Changes

- `removeHTMLTags` replaced with `sanitize`
- `ScriptLoader` replaced with dynamic imports (marked)
- added silentConsoleErrors function

# Tests

- updated all marked related tests

# Notes
Most of the tests expects `<Markdown>` to render raw text in `.fallback` block (also displays console.error), so `silentConsoleErrors` was added to all components that display markdown.

# Screenshots

https://user-images.githubusercontent.com/98811342/165469560-55c02259-e025-446e-9ea3-fb9ced95265e.mov

### CODE REVIEW

File: frontend/svelte/src/lib/services/utils.services.ts [hunk 1]
```diff
@@ -60,3 +62,16 @@ export const queryAndUpdate = async <R, E>({
       .finally(() => (certifiedDone = true)),
   ]);
 };
+
+/**
+ * Sanitize markdown text and convert it to HTML
+ */
+export const markdownToSanitisedHTML = async (
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/services/utils.services.ts:69]: why is this a `service` not a pure util?

File: frontend/svelte/src/lib/utils/markdown.utils.ts [hunk 2]
```diff
@@ -50,3 +50,16 @@ export const renderer = (marked: Marked): Renderer => {
...
   return renderer;
 };
+
+/**
+ * Uses markedjs
+ * @see {@link https://github.com/markedjs/marked}
+ */
+export const markdownToHTML = async (): Promise<(text: string) => string> => {
+  const url = "/assets/assets/libs/marked.min.js";
+  const { marked }: { marked: Marked } = await import(url);
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/utils/markdown.utils.ts:60]: is this going to load every time that we use the util? And rely on the browser cache I guess?

File: frontend/svelte/src/lib/utils/security.utils.ts [hunk 3]
```diff
@@ -1,14 +1,9 @@
...
-  text?: string | undefined
-): string | null | undefined => {
-  if (text === undefined) return text;
-
-  decoder = decoder || document.createElement("div");
-  decoder.innerHTML = text;
-  return decoder.textContent;
+export const sanitize = async (): Promise<(text: string) => string> => {
+  const url = "/assets/assets/libs/purify.min.js";
+  const { sanitize: purify } = (await import(url)).default;
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/utils/security.utils.ts:7]: ditto on importing every time it's used?

File: frontend/svelte/src/lib/components/ui/Markdown.svelte [hunk 4]
```diff
@@ -1,41 +1,26 @@
...
-    on:nnsLoad={onLoad}
-    on:nnsError={onError}
-  />
-{:else if globalMarked !== undefined && text !== undefined}
-  {@html globalMarked?.parse(removeHTMLTags(text) ?? "", {
-    renderer: renderer(globalMarked),
-  })}
 {:else}
-  <!-- fallback text content -->
-  <p data-tid="markdown-text">{text}</p>
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/ui/Markdown.svelte:40]: Why did you remove the `data-tid`? can't we keep it?

**Review by lmuntanerDfinity — COMMENTED** [review]
A few questions

File: frontend/svelte/src/lib/components/ui/Markdown.svelte [hunk 5]
```diff
@@ -1,41 +1,26 @@
 <script lang="ts">
-  import type { marked } from "marked";
-  import ScriptLoader from "../common/ScriptLoader.svelte";
   import Spinner from "./Spinner.svelte";
-  import { removeHTMLTags } from "../../utils/security.utils";
-  import { renderer } from "../../utils/markdown.utils";
-  type Marked = typeof marked;
+  import { markdownToSanitisedHTML } from "../../services/utils.services";
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/ui/Markdown.svelte:3]: typo "sanitize" not "sanitise"

File: frontend/svelte/src/lib/components/ui/Markdown.svelte [hunk 6]
```diff
@@ -1,41 +1,26 @@
...
+  let error: boolean = false;
+  const transform = async (text: string) => {
+    try {
+      html = await markdownToSanitisedHTML(text);
+    } catch (err) {
+      console.error(err);
+      error = true;
+    }
   };
+  $: if (text !== undefined) transform(text);
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/ui/Markdown.svelte:17]: you don't need to handle the promise here?

File: frontend/svelte/src/lib/utils/security.utils.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/utils/security.utils.ts:7]: Do you suggest to cache the function instead?

File: frontend/svelte/src/lib/utils/markdown.utils.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/utils/markdown.utils.ts:60]: no, the browser handles esm modules and load these only when needed - i.e. it will load it the first time and then reuse what was previously downloaded

checkout @mstrasinskis above video

File: frontend/svelte/src/lib/components/ui/Markdown.svelte (hunk 6 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/components/ui/Markdown.svelte:17]: The return parameter is not used here and the error handling is done inside the function.

Comment (mstrasinskis) [reply frontend/svelte/src/lib/components/ui/Markdown.svelte:17]: Would you prefer using `.then` instead of async in the transform function?

File: frontend/svelte/src/lib/components/ui/Markdown.svelte (hunk 4 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/components/ui/Markdown.svelte:40]: Good point! Done.

File: frontend/svelte/src/lib/components/ui/Markdown.svelte (hunk 5 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/components/ui/Markdown.svelte:3]: Thanks, fixed

File: frontend/svelte/src/lib/utils/markdown.utils.ts (hunk 2 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/utils/markdown.utils.ts:60]: Oh! Sorry @mstrasinskis the screen of the laptop was not large enough and I didn't go down the main comment to see the video.

It's ok, I guess we can rely on the browser.

File: frontend/svelte/src/lib/services/utils.services.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/services/utils.services.ts:69]: I tend to avoid having dependencies between utils due to possible circular dependency problems in the future. But I'm don't have strong opinion on this.

File: frontend/svelte/src/lib/components/ui/Markdown.svelte (hunk 6 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/ui/Markdown.svelte:17]: I always subscribe to the promise but know @lmuntanerDfinity sometimes does not (for good reason), so just wanted to know if there was a particular reason here not to, just out of curiosity.

File: frontend/svelte/src/lib/services/utils.services.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/services/utils.services.ts]: refactored
